core/trino-main/src/main/java/io/trino/sql/planner/rowpattern/IrRowPattern.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner.rowpattern;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+@JsonTypeInfo(
+        use = JsonTypeInfo.Id.NAME,
+        property = "@type")
+@JsonSubTypes({
+        @JsonSubTypes.Type(value = IrAnchorPattern.class, name = "anchorPattern"),
+        @JsonSubTypes.Type(value = IrEmptyPattern.class, name = "emptyPattern"),
+        @JsonSubTypes.Type(value = IrExcludedPattern.class, name = "excludedPattern"),
+        @JsonSubTypes.Type(value = IrLabel.class, name = "label"),
+        @JsonSubTypes.Type(value = IrPatternAlternation.class, name = "patternAlternation"),
+        @JsonSubTypes.Type(value = IrPatternConcatenation.class, name = "patternConcatenation"),
+        @JsonSubTypes.Type(value = IrPatternPermutation.class, name = "patternPermutation"),
+        @JsonSubTypes.Type(value = IrQuantifiedPattern.class, name = "quantifiedPattern"),
+})
```

Comment:
You can remove the "Pattern" part from all these class a json type field names:

IrAnchorPattern => IrAnchor, name = "anchor"
IrEmptyPattern => IrEmpty, name = "empty"
IrExcludedPattern => IrExclusion, name ="exclusion"
...
IrPatternPermutation => IrPermutation, name = "permutation"
IrQuantifiedPattern => IrQuantified, name = "quantified"

It's obvious they are all "patterns", so that's redundant.